src/pepr/operator/crd/sources/package/v1alpha1.ts

```diff
@@ -299,8 +299,29 @@ const sso = {
       attributes: {
         description: "Specifies attributes for the client.",
         type: "object",
-        additionalProperties: {
-          type: "string",
+        properties: {
+          "oidc.ciba.grant.enabled": {
+            type: "string",
+          },
+          "backchannel.logout.session.required": {
+            type: "string",
+          },
+          "backchannel.logout.revoke.offline.tokens": {
+            type: "string",
+          },
+          "post.logout.redirect.uris": {
+            type: "string",
+          },
+          "oauth2.device.authorization.grant.enabled": {
+            type: "string",
+          },
+          "pkce.code.challenge.method": {
+            type: "string",
+            enum: ["S256", "plain"],
+          },
+          "client.session.idle.timeout": {
+            type: "string",
+          },
```

Comment:
I'm thinking there might be more value in having this in our Package validator, rather than in the schema. Current behavior with extra properties in `attributes` is that they just get silently dropped (or maybe a warning log message, but still dropped). If we put this in the validate instead we could send a clear message back to the user that certain attributes are not allowed. This would mean we keep the schema/type loose (any string properties allowed).

Either way we should probably find a location to document the allowed attributes so that it's not hidden in the code only.

Reply:
udpated